### PR TITLE
issue #91 - add x509_check_host

### PR DESCRIPTION
**Pull request by mozmark**

### CODE REVIEW

**Comment by mozmark** [comment]
Hi @zhaozg - I'm not clear on why the above CI build is failing. Any suggestions?

**Comment by moteus** [comment]
Problem is `undefined symbol: X509_check_host`
[openssl doc](https://www.openssl.org/docs/manmaster/crypto/X509_check_ip.html)
```
These functions were added in OpenSSL 1.0.2.
```
So I think you need add condition

File: src/x509.c [hunk 1]
```diff
@@ -385,6 +385,24 @@ static LUA_FUNCTION(openssl_x509_check)
   }
 }
 
+static LUA_FUNCTION(openssl_x509_check_host)
+{
+#if OPENSSL_VERSION_NUMBER > 0x10002000L
```
Comment (zhaozg) [inline src/x509.c]: please move #if OPENSSL_VERSION_NUMBER > 0x10002000L out of openssl_x509_check_host,

File: src/x509.c [hunk 2]
```diff
@@ -852,6 +870,7 @@ static luaL_Reg x509_funcs[] =
   {"parse",       openssl_x509_parse},
   {"export",      openssl_x509_export},
   {"check",       openssl_x509_check},
+  {"check_host",  openssl_x509_check_host},
```
Comment (zhaozg) [inline src/x509.c]: please add  #if OPENSSL_VERSION_NUMBER > 0x10002000L before here.

**Comment by zhaozg** [comment]
Thanks, can you add support X509_check_email, X509_check_ip and X509_check_ip_asc.

**Comment by mozmark** [comment]
@zhaozg - I've not added support for X509_check_ip since X509_check_ip_asc provides the same functionality in a form that's more useful to most callers (from lua, at least).

**Pull request merged**